src/main/kotlin/org/rust/lang/core/psi/ext/RsExpr.kt

```diff
@@ -29,13 +29,67 @@ val RsUnaryExpr.operatorType: UnaryOperator get() = when {
     else -> error("Unknown unary operator type: `$text`")
 }
 
+enum class ArithmeticOp(val traitName: String, val itemName: String, val sign: String) {
+    ADD("Add", "add", "+"), // `a + b`
+    SUB("Sub", "sub", "-"), // `a - b`
+    MUL("Mul", "mul", "*"), // `a * b`
+    DIV("Div", "div", "/"), // `a / b`
+    REM("Rem", "rem", "%"), // `a % b`
+    BIT_AND("BitAnd", "bitand", "&"), // `a & b`
+    BIT_OR("BitOr", "bitor", "|"), // `a | b`
+    BIT_XOR("BitXor", "bitxor", "^"), // `a ^ b`
+    SHL("Shl", "shl", "<<"), // `a << b`
+    SHR("Shr", "shr", ">>"); // `a >> b`
+
+    operator fun component1(): String = traitName
+    operator fun component2(): String = itemName
+    operator fun component3(): String = sign
+}
+
 val RsBinaryExpr.operator: PsiElement
     get() = requireNotNull(node.findChildByType(BINARY_OPS)) { "guaranteed to be not-null by parser" }.psi
 
 // TODO: probably want to use a special `enum` here instead of `IElementType`.
 val RsBinaryExpr.operatorType: IElementType
```

Comment:
Hm, Indeed. Yeah, let's create a sealed class then: having two similar ways to do the same thing is a recipe for problems.